tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/Common/Request.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Threading.Tasks;
+using Flurl;
+using Flurl.Http;
+using TechTalk.SpecFlow;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps.Common
+{
+    internal sealed class Request
+    {
+        private readonly Response _response;
+        private readonly ScenarioContext _context;
+
+        public Request(Response response, ScenarioContext context)
+        {
+            _response = response;
+            _context = context;
+        }
+
+        public async Task GetAsync(string url, params object[] pathSegments)
+        {
+            _response.Result = await url
```

Comment:
The first 4 lines in each of these methods are duplicated. Is there a nice way of refactoring so this piece of code is not duplicated. Though I should point out it is still easy to read as it is.

Reply:
😄 Great that you mention this! 

I've pondered that when I was writing this class and...

There is a way, however it is not as pretty.

We can have an extension method that combines those repeated steps.

It would looks something like this, however I think it takes some of the readability away.

![image](https://user-images.githubusercontent.com/37105695/80604200-586c3480-8a29-11ea-9942-ef90e7340b73.png)

If you find that its legible let me know and I can refactor it to ^ and reduce the duplication.

Reply:
Yeah that doesn't read that well. I was thinking more along the lines of:

![image](https://user-images.githubusercontent.com/53512773/80616937-64f88900-8a39-11ea-9f39-0da829992733.png)

Reply:
That looks much better. I'll go with that. Thanks!